src/api/helpers.jl

```diff
@@ -790,6 +811,27 @@ function h5p_get_virtual_view(dapl_id)
     return view[]
 end
 
+"""
+    h5p_get_file_image(fapl_id)::Vector{UInt8}
+
+Retrieve a file image of the appropriate size in a `Vector{UInt8}`.
+"""
+function h5p_get_file_image(fapl_id)::Vector{UInt8}
+    buf_ptr_ref = Ref{Ptr{Nothing}}()
+    buf_len_ref = Ref{Csize_t}(0)
+    h5p_get_file_image(fapl_id, buf_ptr_ref, buf_len_ref)
+    unsafe_wrap(Array{UInt8}, Ptr{UInt8}(buf_ptr_ref[]), buf_len_ref[]; own=true)
```

Comment:
I'm always a little bit worried about `own=true`: I know the docs say to call `free`, but I thought that this wasn't correct on Windows?

Reply:
It might be safer to do the double call: first call with `buf_ptr_ref` as `C_NULL` to get the size of the buffer, then allocate it ourselves, and call it again?

Reply:
Oh, i was wrong. You would need to define [`H5Pset_file_image_callbacks`](https://docs.hdfgroup.org/hdf5/v1_12/group___f_a_p_l.html#ga14ea3598215afd078b964b672b40d63c)

Reply:
What's strange is that HDF5 explicitly provides [`H5_FREE_MEMORY`](https://portal.hdfgroup.org/display/HDF5/H5_FREE_MEMORY) for this reason elsewhere (see #677).

Reply:
We probably should use that. I'll check the HDF5 source later.

Reply:
I'm resolving this for now by only allowing the one-arg version with `own=true` if `h5p_get_file_image_callbacks` returns a null pointer for `image_free`.

Reply:
Maybe we should still use `HDF5.API.h5_free_memory`.